docs/_guides/mango-queries.md

```diff
@@ -0,0 +1,330 @@
+
+This index may be good for answering questions like "find all 17-year-olds whose name starts with letters N-Z", but it's not very good for answering questions like "find all people with a certain name, older than a certain age."
+
+This shows that it's important to carefully design an index before creating a query to use that index. Otherwise, the query planner may fall back to in-memory querying, which can be expensive.
+
+{% include anchor.html title="Performance notes" hash="performance-notes" %}
+
+The Mango query language is generally very permissive, and allows you to write queries that may not perform very well, but will run regardless. For instance, you may create an index with `createIndex()`, but then write a `find()` query that doesn't actually use that index. In general, the query planner tries to find the most appropriate index, but it may fall back to in-memory querying.
+
+As a straightforward example, if you query using the `_id` field, then the query planner will automatically map that directly to an `allDocs()` query. However, if you query for a field that isn't yet indexed, then it will simply use `allDocs()` to read in all documents from the database (!) and then filter in-memory. This can lead to poor performance, especially if your database is large.
+
+If you're ever wondering how the query planner is interpreting your query, you can enable debugging like so:
+
+```js
+PouchDB.debug.enable('pouchdb:find');
+```
+
+In the console, the query planner will show a detailed explanation of how it has interpreted the query, whether it uses any indexes, and whether any parts of the query need to be executed in-memory.
+
+You may also want to pay attention to the `"warning"` value included in your results set, indicating that there was no index that matched the given query. For instance, the warning may look like this:
+
+```js
+{
+  "docs": [ /* ... */ ],
+  "warning": "no matching index found, create an index to optimize query time"
+}
+```
+
+{% include anchor.html title="Further reading" hash="further-reading" %}
+
+The Mango query language is quite large and supports many options. Some of the more common ones include:
+
+* `$eq`: equals
+* `$gt`: greater than
+* `$gte`: greater than or equal to
+* `$lt`: less than
+* `$lte`: less than or equal to
+
+There are many more options besides these, although note that not all of them can take advantage of indexes. For instance, `$regex`, `$ne`, and `$not` cannot use on-disk indexes, and must use in-memory filtering instead.
+
+The most complete documentation for selector options can be found at [Cloudant Query Language](https://docs.cloudant.com/cloudant_query.html) (which is the same thing as Mango). Any differences between this specification and PouchDB's implementation should be considered a bug; they are designed to be identical!
```

Comment:
I think the [CouchDB documentation](http://docs.couchdb.org/en/2.0.0/api/database/find.html) is cleaner then Cloudant's; the Cloudant documentation also covers the `text` index type which isn't supported by CouchDB and, in my opinion, confuses things considerably.